### PR TITLE
Created delete_RGB function.

### PR DESCRIPTION
**Pull request by Cyberzoid1**
Need review

Todo:
Testing
Integration into FrameList Delete

### CODE REVIEW

**Comment by gjcarter** [comment]
Looks good to me. Should be very simple to integrate it into FrameList Delete. Would look something like this.

```/*Deletes the first node in the list*/
void FrameList::DeleteNode(){
    NodePtr p = head;

    if( p == NULL){
        /*Nothing. Error case.*/
    }
    else{
        head = p->next;
        p->next = NULL;
        for(int i = 0; i < p->FrameData.r; i++)
            delete p->FrameData.data[i];
        delete p->FrameData.data;
        p->FrameData.data = nullptr;
        delete p;
    }
    this->count--;
}
```

**Comment by Cyberzoid1** [comment]
I added the delete_RGB(..) function to the FrameList delete methods

However I ocassionaly am getting a segfault when using the run-debug. It tends to run with normal run.

Can someone verify and look over my integration with Framelist?

File: GoofyLightsEditor/FrameList.cpp [hunk 1]
```diff
@@ -116,8 +117,7 @@ void FrameList::DeleteNode_Middle(int pos){
         // Delete the head node.
         head = current -> next;
         head -> prev = NULL;
-        // Delete Attached RGB structure here 
-        delete(current);
```
Comment (Kevin-Dorscher) [inline GoofyLightsEditor/FrameList.cpp:120]: This is the only issue I spot at the moment, you delete the RGB structure, but erased the line that deleted that node in the linked list. I'll have to look over this some more, but at this point it looks like the RGB structure has been freed, but not the node in the linked list that was connected to the deleted RGB structure.

Comment (Kevin-Dorscher) [reply GoofyLightsEditor/FrameList.cpp:120]: So i think:
delete_RGB(head -> FrameData.data, this -> row, this -> col);
delete (current);

would fix this issue, this my also be what is causing the seg fault, which would only happen when calling this function.

Comment (Cyberzoid1) [reply GoofyLightsEditor/FrameList.cpp:120]: Fixed

**Comment by Cyberzoid1** [comment]
I Believe this is ready.

**Pull request merged**